.github/workflows/release.yaml

```diff
@@ -0,0 +1,41 @@
+name: Release
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+  id-token: write
+
+env:
+  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+jobs:
+  release:
+    name: Release Package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Use Node.js 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - name: Set up .npmrc
+        run: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
+      - name: Install Dependencies
+        run: npm install
+      - name: Linter Check
+        run: npm run lint
+      - name: Run Tests
+        run: npm run test
+      - name: Run Build
+        run: npm run build
+      - name: Publish to npm
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npm publish --access public
```

Comment:
the release process should handle the package version. consider using symantic releases like we did with hyphen components

Reply:
@osbornm  will the release be made automatically when a push is made to the main branch, or will it be done manually using GitHub Releases like the CLI?

Reply:
I'd like to handle this exactly like hyphen components does. Please look at how it handles releases and how it updates package.json as well.

Reply:
another good example https://github.com/Hyphen/nucleus?tab=readme-ov-file#releases